react-native-sdk/android/src/main/java/org/jitsi/meet/sdk/JMOngoingConferenceService.java

```diff
@@ -0,0 +1,184 @@
+
+    static final int NOTIFICATION_ID = new Random().nextInt(99999) + 10000;
+
+    public static void doLaunch(Context context) {
+        try {
+            RNOngoingNotification.createOngoingConferenceNotificationChannel(context);
+            JitsiMeetLogger.w(TAG + " Notification channel creation completed");
+        } catch (Exception e) {
+            JitsiMeetLogger.e(e, TAG + " Error creating notification channel");
+        }
+
+        Intent intent = new Intent(context, JMOngoingConferenceService.class);
+
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                context.startForegroundService(intent);
+                JitsiMeetLogger.w(TAG + " Starting foreground service");
+            } else {
+                context.startService(intent);
+                JitsiMeetLogger.w(TAG + " Starting service");
+            }
+        } catch (RuntimeException e) {
+            // Avoid crashing due to ForegroundServiceStartNotAllowedException (API level 31).
+            // See: https://developer.android.com/guide/components/foreground-services#background-start-restrictions
+            JitsiMeetLogger.w(TAG + " Ongoing conference service not started", e);
+        }
+    }
+
+    public static void launch(Context context, ReactActivity reactActivity) {
+
+        PermissionListener listener = new PermissionListener() {
+            @Override
+            public boolean onRequestPermissionsResult(int i, String[] strings, int[] results) {
+                JitsiMeetLogger.w(TAG + " Permission callback received");
+
+                if (results == null || results.length == 0) {
+                    JitsiMeetLogger.w(TAG + " Permission results are null or empty");
+                    return false;
+                }
+
+                int counter = 0;
+                for (int result : results) {
+                    if (result == PackageManager.PERMISSION_GRANTED) {
+                        counter++;
+                    }
+                }
+
+                JitsiMeetLogger.w(TAG + " Permissions granted: " + counter + "/" + results.length);
+
+                if (counter == results.length) {
+                    JitsiMeetLogger.w(TAG + " All permissions granted, launching service");
+                    doLaunch(context);
+                    return true;
+                } else {
+                    JitsiMeetLogger.w(TAG + " Not all permissions were granted");
+                    return false;
+                }
+            }
+        };
+
+        List<String> permissionsList = new ArrayList<>();
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            JitsiMeetLogger.w(TAG + " Checking Tiramisu permissions");
+            permissionsList.add(POST_NOTIFICATIONS);
+            permissionsList.add(RECORD_AUDIO);
+        }
+
+        String[] permissionsArray = new String[ permissionsList.size() ];
```

Comment:
Put all the permissions dance inside the `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {` if block, since you don't need any of it otherwise.